app/views/trips/_form.html.haml

```diff
@@ -108,8 +108,7 @@
     }).addTo(myMap);
 
     var initialTripPoints = #{raw @trip.points.sort { |x, y| x.rank <=> y.rank }.map { |point| [point.lat.to_f, point.lon.to_f]}};
-    window.myTripDrawing = new TripDrawing().init(myRouting, initialTripPoints);
-    window.myRouting = myRouting;
+    var myTripDrawing = new TripDrawing().init(myRouting, initialTripPoints);
```

Comment:
Might separate definition and init and by adding parenthesis like `(new TripDrawing()).init`

Reply:
var myTripDrawing = (new TripDrawing()).init(myRouting, initialTripPoints);

like this ? I don't get it sorry...

Reply:
The flow is more easy to understand.
Generally we use tow different lines.

Reply:
like this ?
var myTripDrawing = (new TripDrawing())
  .init(myRouting, initialTripPoints);

Reply:
more
```
var myTripDrawing = (new TripDrawing())
myTripDrawing.init(myRouting, initialTripPoints);
```

or 
```
var myTripDrawing = (new TripDrawing())
var myTripDrawingInit = myTripDrawingInit.init(myRouting, initialTripPoints);
```

if needed

to avoid operator precedence miss comprehension : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence

and they are at the same priority level ! so the result could be different according to JS interpretor implementation
- new (with argument list) | n/a | new … ( … )
- Function Call | left-to-right | … ( … )

Example in Chrome :

```
> new Date().toISOString()
"2017-10-23T21:37:27.780Z"
// Priority 19 and 19

> new Date.toISOString()
VM114:1 Uncaught TypeError: Date.toISOString is not a constructor
    at <anonymous>:1:1
(anonymous) @ VM114:1
// Priority 18 and 19

> (new Date).toISOString()
"2017-10-23T21:37:49.444Z"
// Priority 20 and 19
```

Reply:
Okidoki !
Thanks for all the explanations !!